### PR TITLE
Update app and issues workflow

### PR DESCRIPTION
**Pull request by sowbiba**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,40 +28,58 @@ const IssueDataProvider = require('./src/issue-data-provider.js');
...
+        todo: { name: 'To Do', automatic: true },
+        toBeReproduced: { name: 'TBR', automatic: true },
+        toBeSpecified: { name: 'TBS', automatic: true },
+        needsMoreInfo: { name: 'NMI', automatic: true },
+        fixed: { name: 'Fixed', automatic: true }
       },
       milestones: {
         next_patch_milestone: '1.7.4.3',
-        next_minor_milestone: '1.7.5.0',
+        next_minor_milestone: '1.7.5.0'
```
Comment (matks) [inline index.js]: These are very outdated but can be fixed later

**Comment by matks** [comment]
@sowbiba Looks good to me. Let's see if @eternoendless has something to add, else it's OK

@PierreRambaud and @NeOMakinG can also help to review Javascript

File: index.js [hunk 2]
```diff
@@ -28,40 +28,58 @@ const IssueDataProvider = require('./src/issue-data-provider.js');
...
 
-    let rulePromise = ruleComputer.findRule(context);
-    let rule = await rulePromise;
+    const rulePromise = ruleComputer.findRules(context);
+    const rules = await rulePromise;
 
-    if (rule != null) {
-      context.log.info('[Index] Received webhook matches rule ' + rule + ' requirements');
-      ruleApplier.applyRule(rule, context);
+    if (rules != null) {
```
Comment (NeOMakinG) [inline index.js]:
```suggestion
    if (rules !== null) {
```

File: src/issue-data-provider.js [hunk 3]
```diff
@@ -104,7 +106,7 @@ module.exports = class IssueDataProvider {
 
     const allCards = [];
     await Promise.all(vars.map(async (value) => {
-      const cards = await this.githubApiClient.projects.getProjectCards({column_id: this.config.kanbanColumns[value]});
+      const cards = await this.githubApiClient.projects.listCards({column_id: this.config.kanbanColumns[value]});
       allCards.push(cards.data);
```
Comment (NeOMakinG) [inline src/issue-data-provider.js]:
```suggestion
```

File: src/rule-applier.js [hunk 4]
```diff
@@ -124,10 +151,86 @@ module.exports = class RuleApplier {
...
+
+        await this.githubApiClient.issues.update({
+          issue_number: issueId,
+          owner: context.payload.repository.owner.login,
+          repo: context.payload.repository.name,
+          state: 'open'
+        })
+      }
+    }
+  }
```
Comment (NeOMakinG) [inline src/rule-applier.js]: This looks so duplicating, wdyt @PierreRambaud ?

File: index.js (hunk 2 above)
Comment (NeOMakinG) [reply index.js]: Also is if(rules) not enough?

File: package.json [hunk 5]
```diff
@@ -16,27 +16,36 @@
...
   },
   "dependencies": {
-    "probot": "^7.0.0"
+    "probot": "^9.6.4"
   },
   "devDependencies": {
-    "jest": "^22.4.3",
-    "nodemon": "^1.17.2",
-    "smee-client": "^1.0.2"
+    "jest": "^24.9.0",
```
Comment (PierreRambaud) [inline package.json]: Don't hesitate to add our eslint configuration :)

File: index.js [hunk 6]
```diff
@@ -28,40 +28,58 @@ const IssueDataProvider = require('./src/issue-data-provider.js');
...
+      for (const rule of rules) {
```
Comment (PierreRambaud) [inline index.js]: The for statement is also done in `applyRules`, is it needed to do both?

File: src/rule-applier.js [hunk 7]
```diff
@@ -40,33 +40,60 @@ module.exports = class RuleApplier {
...
-
-      default:
-        this.logger.error('[Rule Applier] Cannot apply ' + rule);
+  applyRules (rules, context) {
+    for (const rule of rules) {
+      this.logger.debug('[Rule Applier] Applying rule ' + rule);
+
+      switch (rule) {
+        case Rule.A1:
+          this.applyRuleA1(context);
```
Comment (PierreRambaud) [inline src/rule-applier.js]: Should be better to find a way to remove this useless switch, otherwise in few weeks, we'll have a 500 lines only for the switch.

File: src/rule-applier.js (hunk 4 above)
Comment (PierreRambaud) [reply src/rule-applier.js]: Yep :+1:

Comment (NeOMakinG) [reply src/rule-applier.js]: Then would it be possible to try to do something to avoid duplicate code ? Sometimes  it's too overkill but there it looks like not

File: .eslintignore [hunk 8]
```diff
@@ -0,0 +1,3 @@
+/node_modules/
+/tests/
+!.eslintrc.js
```
Comment (NeOMakinG) [inline .eslintignore]: Please configure your IDE so it auto-add new line at the end of files!

File: .eslintrc.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
+// http://eslint.org/docs/user-guide/configuring
+
+module.exports = {
+  root: true,
+  env: {
+    browser: true,
+    node: true,
+    es6: true,
+    jquery: true,
```
Comment (NeOMakinG) [inline .eslintrc.js]: No need this on this project right?

File: .eslintrc.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  globals: {
+    google: true,
+    document: true,
+    navigator: false,
+    window: true,
```
Comment (NeOMakinG) [inline .eslintrc.js]: No need too, it's node side, not client side right ?

File: .eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    'max-len': ['error', {code: 120}],
+    'no-alert': 0,
+    'no-bitwise': 0,
+    'no-new': 0,
+    'no-param-reassign': ['error', {props: false}],
+    'no-restricted-globals': [
+      'error',
+      {
+        name: 'global',
+        message: 'Use window variable instead.',
```
Comment (NeOMakinG) [inline .eslintrc.js]: It's node side, you'll never use window

File: src/pull-request-data-provider.js [hunk 12]
```diff
@@ -22,19 +22,18 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  * International Registered Trademark & Property of PrestaShop SA
  */
-const { graphql } = require("@octokit/graphql");
-const { createAppAuth } = require("@octokit/auth-app");
+const {graphql} = require('@octokit/graphql');
```
Comment (NeOMakinG) [inline src/pull-request-data-provider.js]: I strongly disagree with the fact that we use octokit library as there was no updates since 4 months and apollo is strongly maintained/used in the JS community, we can still use https://github.com/apollographql/apollo-link#readme with a fetch polyfill !

Wdyt @PierreRambaud @eternoendless 

I know that's not related to this PR, but it's pretty weird to be honest

File: src/pull-request-data-provider.js [hunk 13]
```diff
@@ -134,49 +131,49 @@ module.exports = class PullRequestDataProvider {
...
+        `{
             repository(owner: "${owner}", name: "${repo}") {
               pullRequest(number: ${pullRequestId}) {
                 bodyHTML
               }
             }
           }
-        `
+        `,
       );
```
Comment (NeOMakinG) [inline src/pull-request-data-provider.js]: A good improve would be to extract these queries like a pageMap file and use them importing a JSON object, but well it looks not related to this PR too...

File: src/rule-applier.js [hunk 14]
```diff
@@ -22,32 +22,30 @@
...
+const G2 = require('./rules/G2.js'); // eslint-disable-line no-unused-vars
+const H1 = require('./rules/H1.js'); // eslint-disable-line no-unused-vars
+const H2 = require('./rules/H2.js'); // eslint-disable-line no-unused-vars
+const I1 = require('./rules/I1.js'); // eslint-disable-line no-unused-vars
+const J1 = require('./rules/J1.js'); // eslint-disable-line no-unused-vars
+const J3 = require('./rules/J3.js'); // eslint-disable-line no-unused-vars
+const J4 = require('./rules/J4.js'); // eslint-disable-line no-unused-vars
+const K1 = require('./rules/K1.js'); // eslint-disable-line no-unused-vars
+const L1 = require('./rules/L1.js'); // eslint-disable-line no-unused-vars
+const L2 = require('./rules/L2.js'); // eslint-disable-line no-unused-vars
```
Comment (NeOMakinG) [inline src/rule-applier.js]:
```suggestion
/* eslint-disable no-unused-vars */
const A1 = require('./rules/A1.js'); 
const B2 = require('./rules/B2.js');
const C1 = require('./rules/C1.js'); 
const C2 = require('./rules/C2.js');
const D1 = require('./rules/D1.js');
const D2 = require('./rules/D2.js');
const D3 = require('./rules/D3.js');
const D4 = require('./rules/D4.js');
const E1 = require('./rules/E1.js');
const E3 = require('./rules/E3.js');
const E4 = require('./rules/E4.js');
const E5 = require('./rules/E5.js');
const E6 = require('./rules/E6.js');
const F1 = require('./rules/F1.js');
const G2 = require('./rules/G2.js');
const H1 = require('./rules/H1.js');
const H2 = require('./rules/H2.js');
const I1 = require('./rules/I1.js');
const J1 = require('./rules/J1.js');
const J3 = require('./rules/J3.js');
const J4 = require('./rules/J4.js');
const K1 = require('./rules/K1.js');
const L1 = require('./rules/L1.js');
const L2 = require('./rules/L2.js'); 
/* eslint-enable */
```

File: src/pull-request-data-provider.js (hunk 12 above)
Comment (sowbiba) [reply src/pull-request-data-provider.js]: Probot is already using octokit/graphql
https://github.com/probot/probot/blob/master/package.json#L55

But I can change our used lib to apollographql

File: .eslintrc.js (hunk 9 above)
Comment (sowbiba) [reply .eslintrc.js]: You right. It seems not needed. I will check what every parameter is for in `env` and `globals`

File: index.js (hunk 2 above)
Comment (sowbiba) [reply index.js]: done

File: src/issue-data-provider.js (hunk 3 above)
Comment (sowbiba) [reply src/issue-data-provider.js]: done

File: package.json (hunk 5 above)
Comment (sowbiba) [reply package.json]: done

File: index.js (hunk 6 above)
Comment (sowbiba) [reply index.js]: done

File: src/rule-applier.js (hunk 7 above)
Comment (sowbiba) [reply src/rule-applier.js]: I tried to do it better

File: src/pull-request-data-provider.js (hunk 13 above)
Comment (sowbiba) [reply src/pull-request-data-provider.js]: As we have variables in the json object, I dont know how to do it by importing a JSON file

File: .eslintrc.js (hunk 9 above)
Comment (PierreRambaud) [reply .eslintrc.js]: Env is for context, globals is for globals variables.

File: index.js (hunk 2 above)
Comment (PierreRambaud) [reply index.js]: I don't see the change in file :thinking:

File: src/pull-request-data-provider.js (hunk 12 above)
Comment (NeOMakinG) [reply src/pull-request-data-provider.js]: It's not related to the PR, wait for @eternoendless and @PierreRambaud advice

File: src/pull-request-data-provider.js (hunk 13 above)
Comment (NeOMakinG) [reply src/pull-request-data-provider.js]: easily doable with apollo but I don't know if you can do it with octokit

File: src/pull-request-data-provider.js (hunk 12 above)
Comment (PierreRambaud) [reply src/pull-request-data-provider.js]: @NeOMakinG `@octokit/graphql` has been updated one week ago on GitHub whereas apollo was one month ago. Not having a release doesn't mean the project is not maintained. And because probot is also using this library, we should reuse the same library :thinking:

Comment (NeOMakinG) [reply src/pull-request-data-provider.js]: Well I should have express that differently, Apollo is a reference used by most of JS community members and maintained by 10x more customers than octokit, but ok.

File: src/config-provider.js [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+   */
+  constructor(config, githubApiClient, issueDataProvider, projectDataProvider, logger) {
+    this.config = config;
+    this.githubApiClient = githubApiClient;
+    this.issueDataProvider = issueDataProvider;
+    this.projectDataProvider = projectDataProvider;
+    this.logger = logger;
+  }
+
+  getRepositoryConfigFromMilestone(config, milestone) {
```
Comment (eternoendless) [inline src/config-provider.js]: Consider adding descriptions for methods as well as for parameters, including types and return types

This helps A LOT to understand code :)

Comment (sowbiba) [reply src/config-provider.js]: done ;)
